pinot-core/src/main/java/org/apache/pinot/core/operator/combine/SelectionOrderByCombineOperator.java

```diff
@@ -76,8 +77,10 @@ protected IntermediateResultsBlock getNextBlock() {
     assert orderByExpressions != null;
     if (orderByExpressions.get(0).getExpression().getType() == ExpressionContext.Type.IDENTIFIER) {
       try {
-        return new MinMaxValueBasedSelectionOrderByCombineOperator(_operators, _queryContext,
-            _executorService).getNextBlock();
+        return new MinMaxValueBasedSelectionOrderByCombineOperator(_operators, _queryContext, _executorService)
+            .getNextBlock();
+      } catch (QueryCancelledException e) {
+        throw new QueryCancelledException("Cancelled while running MinMaxValueBasedSelectionOrderByCombineOperator", e);
```

Comment:
(minor) We can simply throw it out because the stack trace should have the original class info
```suggestion
        throw e;
```